charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml

```
@@ -47,6 +47,12 @@ spec:
   logFormat: {{ .Values.alertmanager.alertmanagerSpec.logFormat | quote  }}
   logLevel:  {{ .Values.alertmanager.alertmanagerSpec.logLevel | quote  }}
   retention: {{ .Values.alertmanager.alertmanagerSpec.retention | quote  }}
+  {{- if .Values.alertmanager.enableFeatures }}
+  enableFeatures:
+    {{- range .Values.alertmanager.enableFeatures }}
+    - {{ . | quote }}
+    {{- end }}
+  {{- end }}
```

this should work, too?

```suggestion
  {{- with .Values.alertmanager.enableFeatures }}
  enableFeatures:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

---

sure it would.

---

> sure it would, but if the features are listed without quotes, that is why I made use of quotes, but no worries if you want I can change it and in case of `{{- with }}` it completely slipped of my mind.

---

Could you explain, what you mean with the quotes? Do I miss something?

---

I meant if there is something like this in values.yaml

```
alertmanager:
  enableFeatures:
    - feature1
    - "feature2"
```
using quote would ensure the template renders it correctly as string, like this:

```
enableFeatures:
  - "feature1"
  - "feature2"
```

---

Ah, I see. Quotes are most of the time optional. They are only required, if strings are expected, but numbers are given.

with toYaml, 

```
alertmanager:
  enableFeatures:
    - feature1
    - "feature2"
```

would result into:

```
alertmanager:
  enableFeatures:
    - feature1
    - feature2
```

---

yes that's right its just slipped off my mind to add with toYam